### PR TITLE
Migrate to mamba-org/setup-micromamba

### LINKED ISSUE #1746: replace `provision-with-micromamba`
The `provision-with-micromamba` github action is deprecated. See migration examples to `setup-micromamba` [here](https://github.com/mamba-org/provision-with-micromamba#migration-to-setup-micromamba). 

This is the only spot we use it in our CI:

https://github.com/pvlib/pvlib-python/blob/30c62e368529df01faa609d6b38456a7b0db9b53/.github/workflows/pytest.yml#L43

### PR DESCRIPTION
**Pull request by d-watt**
- [x] Closes #1746 
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

Migrating to mamba-org/setup-micromamba, as outlined in #1746

### CODE REVIEW

**Comment by d-watt** [comment]
@wholmgren since it's my first pvlib pr, looks like some workflows require approval to trigger? I've taken a shot at addressing #1758, ready to review!

**Comment by kandersolar** [comment]
Looks good to me, thanks @d-watt!  Can you also make a brief note under `Testing` and add yourself to the contributors list in https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/whatsnew/v0.9.6.rst?

**Comment by d-watt** [comment]
Thanks @kandersolar! Missed that in the contributing guidelines. Ready to re-review.

> Looks good to me, thanks @d-watt! Can you also make a brief note under `Testing` and add yourself to the contributors list in https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/whatsnew/v0.9.6.rst?

**Comment by d-watt** [comment]
Alright @kandersolar, ready for another look!

File: docs/sphinx/source/whatsnew/v0.9.6.rst [hunk 1]
```diff
@@ -81,4 +81,5 @@ Contributors
 * Andy Lam (:ghuser:`@andylam598`)
 * :ghuser:`ooprathamm`
 * Kevin Anderson (:ghuser:`kandersolar`)
+* Devon Watt (:guser:`d-watt`)
```
Comment (kandersolar) [inline docs/sphinx/source/whatsnew/v0.9.6.rst:84]:
```suggestion
* Devon Watt (:ghuser:`d-watt`)
```

**Review by kandersolar — APPROVED** [review]
LGTM.  Thanks @d-watt!

**Pull request merged**